packages/runtime/src/next/client.ts

```diff
@@ -202,6 +208,32 @@ export class Makeswift {
     return json
   }
 
+  async getPage(
+    pathname: string,
+    { locale: localeInput }: { locale?: string } = {},
+  ): Promise<GetPageResult | null> {
+    const siteVersion = MakeswiftSiteVersion.Live
+
+    const searchParams = new URLSearchParams()
+    if (localeInput) {
+      searchParams.set('locale', localeInput)
+    }
+
+    const response = await this.fetch(
+      `v1/pages/${encodeURIComponent(pathname)}/meta?${searchParams.toString()}`,
+      { headers: { 'Makeswift-Site-Version': siteVersion } },
+    )
+
+    if (!response.ok) {
+      console.error('Failed to get page', await response.json())
+      throw new Error(`Failed to get page with error: "${response.statusText}"`)
+    }
```

Comment:
Can we design it so that our API always returns JSON such that we can always do `reponse.json` and then based on the status code we can get strong type safety for the possible errors and have good error messages here?

Reply:
It might be harder to do that just for this API, because the error could also come from `ThrottlerByApiKeyGuard`, `SiteIdByApiKeyPipe`, or `ParseSiteVersionPipe`. We need to also refactor those if we want our API to always return JSON right?

Reply:
In those scenarios, won't we know the exact status codes? Also, maybe we can update those guards and pipes so that the error is JSON as well?

Reply:
Yeah, but I don't think I'm planning on updating those guards and pipes in this MR. I added a linear issue for this.